### DISCUSSION TITLE
Backup running instance?

### DISCUSSION THREAD

**Discussion by yaakovfeldman** [jellyfin/jellyfin#11019] [Q&A] [answered]
Is there any way to run a backup without stopping Jellyfin first?

I’m running Jellyfin in docker and I have mounted and am backing up the config directory. The only problem is that the sqlite files jellyfin.db and library.db could be in an inconsistent/corrupted state if they are in use at the time. So most backup techniques seem to recommend stopping Jellyfin before running the backup. 

Is there any other way to backup?

I would normally just backup the sqlite files using sqlite3 CLI but Jellyfin opens library.db with an exclusive lock that prevents sqlite3 running backup. (As also found by https://github.com/jellyfin/jellyfin/issues/8478 )

**Comment by yaakovfeldman** [comment]
I assume the same question would apply to other installation methods and operating systems too, not just docker.

**Comment by pedrojreis** [comment, marked as answer]
It's always risky to try and backup something that has an exclusive lock - specially a database.

I believe the most risk free for Jellyfin in docker is to use [volumes](https://docs.docker.com/storage/volumes/) to get that config in the host and then try and copy the files to another path and then backup. But if Jellyfin writes into the DB while you are copying you may get a corrupted backup.

I wouldn't try to make something that was not meant for. Just use some script (cron job) that stops the container, back it up and then starts the container. I believe you are not using jellyfin 24/7 and need 99.99% SLA :D (If so, go ahead and use k8s for availability).
